### PR TITLE
Add golangci-lint fixer

### LINKED ISSUE #4616: Add fixer for `golangci-lint`
**Name:** `golangci-lint`
**URL:** https://github.com/golangci/golangci-lint

This was previously reported at https://github.com/dense-analysis/ale/issues/2864:

> GoLangCI-lint has a fixer now: `golangci-lint run --fix ./...`
>
> The ALE fixer will have the same issues we're having with the ALE linter.

This original issue was to add a fixer for `golangci-lint` and there may have been some confusion as the issue was closed on account of `golangci-lint` as a _linter_ already existing.

As best I can tell and from what I've tried, `golangci-lint` isn't available as a fixer (e.g. it's not visible in `:ALEFixSuggest` nor has a file listed at https://github.com/dense-analysis/ale/tree/master/autoload/ale/fixers or visible elsewhere in the codebase).

### PR DESCRIPTION
**Pull request by sigmavirus24**
Closes #4616

### CODE REVIEW

File: autoload/ale/fixers/golangci_lint.vim [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+
+    if l:package_mode
+        return l:env . ale#Escape(l:executable)
+        \   . ' run '
+        \   .  l:options
+    endif
+
+    return l:env . ale#Escape(l:executable)
+    \   . ' run'
```
Comment (hsanson) [inline autoload/ale/fixers/golangci_lint.vim:23]: Leave an space after after run or the options will be concatenated directly to the run command itself:

```suggestion
    \   . ' run '
```

File: test/fixers/test_golangci_lint_fixer_callback.vader [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+Execute(The golangci-lint callback should support per-file mode):
+  let g:ale_go_golangci_lint_package = 0
+
+  AssertEqual
+  \ {
+  \   'command': ale#Escape('xxxinvalid')
+  \     . ' run'
+  \     . g:ale_go_golangci_lint_options
+  \     . ' --fix testfile.go',
```
Comment (hsanson) [inline test/fixers/test_golangci_lint_fixer_callback.vader:46]: You need to escape the testfile.go here or the test equal assert won't match.

Comment (hsanson) [reply test/fixers/test_golangci_lint_fixer_callback.vader:46]:
```
Starting Vader: /testplugin/test/fixers/test_golangci_lint_fixer_callback.vader
    (1/3) [EXECUTE] The golangci-lint callback should return the correct default values
    (2/3) [EXECUTE] The golangci-lint callback should include custom golangci-lint options
    (3/3) [EXECUTE] The golangci-lint callback should support per-file mode
    (3/3) [EXECUTE] (X) {'command': '''xxxinvalid'' run --fix ''testfile.go'''} should be equal to {'command': '''xxxinvalid'' run --fix testfile.go'}
  Success/Total: 2/3
```

**Review by hsanson — CHANGES_REQUESTED** [review]
There is a failing test.

**Review by hsanson — APPROVED** [review]
Thanks for the fixes.

**Comment by sigmavirus24** [comment]
Thanks for accepting this and helping work through it. This is my first vim plug-in contribution ever

**Pull request merged**